h2o-core/src/main/java/hex/segments/LocalSequentialSegmentModelsBuilder.java

```diff
@@ -54,8 +55,29 @@ void buildModels(SegmentModels segmentModels) {
         SegmentModels.SegmentModelResult result = segmentModels.addResult(segmentIdx, builder, failure);
         Log.info("Finished building a model for segment id=", segmentIdx, ", result: ", result);
         cleanUp(builder);
+        if (result.isSuccessful())
+          stats._succeeded++;
+        else
+          stats._failed++;
       }
     }
+    return stats;
+  }
+
+  static class SegmentModelsStats extends Iced<SegmentModelsStats> {
+    int _succeeded;
```

Comment:
As it's only local, we don't need to take care about atomicity of operations on top of these.